scripts/install.js

```diff
@@ -45,13 +47,9 @@ function applyProxy(options, cb) {
     var proxyUrl;
 
     if (!er) {
-      ['https-proxy', 'proxy', 'http-proxy'].some(function(setting) {
-        var npmProxyUrl = conf.get(setting);
-        if (npmProxyUrl) {
-          proxyUrl = npmProxyUrl;
-          return true;
-        }
-      });
+      proxyUrl = ['https-proxy', 'proxy', 'http-proxy'].filter(function(setting) {
```

Comment:
Yes! underscore definitely provides a great deal of functions when it comes to Array. We can include it in njs projects, but nativity and [vanilla.js](http://vanilla-js.com/) is something I always prefer. :smiley: 

Thanks to node.js to rely on one JS runtime and does not present JS incompatibility issues like those in cross-browser scenarios.